dissect/target/plugins/apps/browser/browser.py

```diff
@@ -93,3 +112,19 @@ def try_idna(url: str) -> bytes:
         return url.encode("idna")
     except Exception:
         return url
+
+
+def keychain_passwords() -> set:
+    """Retrieve a set of passphrases to use for decrypting saved browser credentials.
+
+    Returns: Set of passphrase strings.
+    """
+    keys = keychain.get_keys_for_provider("browser") + keychain.get_keys_without_provider()
```

Comment:
Maybe an idea to make this a class method on the `BrowserPlugin`? Could get the keys for both `browser` and `cls.__namespace__`.

Reply:
Please correct me if I'm missing something, but this is how I currently understand things:

- Firefox uses its own crypto with an optional master password
  - This could use the `browser`/`firefox` keychain provider, optionally with maybe the profile ID as identifier to speed things up (if it's known)
    - If a password for a known profile ID exists, there's no need to "bruteforce" all passwords
  - This requires no interaction with `DPAPIPlugin`
  - The reason why I suggested the class method is so that you could support grabbing keys for both the `browser` provider and `cls.__namespace__`, i.e. `firefox` provider easily. 
    - It's mostly synthetic sugar and allows for a more specific notation of which password belongs to what in the keychain ("Hey which password was the firefox one again?") while also allowing for a slightly specific "browser" wildcard, which benefits other code that uses the keychain (one less key without a provider).
- Chrome on Windows uses DPAPI which uses the Windows user password
  - This password is technically unrelated to any browser, so it should just be a `user`/`dpapi` provider with maybe the user SID as identifier to speed things up (if it's known)
    - If a password for a known SID exists, there's no need to "bruteforce" all passwords
  - This requires no (relevant) crypto on the browser side, so no interaction with the `BrowserPlugin`

The only thing I'm puzzled on is the SHA1 hash thing. Is that a DPAPI or Chrome thing?